ui/src/components/WMWorkerPoolsTable/index.jsx

```diff
@@ -57,6 +57,8 @@ export default class WorkerManagerWorkerPoolsTable extends Component {
     sortDirection: 'asc',
     error: null,
     actionLoading: false,
+    dialogOpen: false,
+    workerPool: null,
```

Comment:
Instead of `workerPool`, `dialogOpen`, and `error`, let's try using `dialogState`:

```jsx
dialogState: {
  error: null,  
  open: false,
  title: '',
  confirmText: '',
  item: null,
},
```

Then we a user clicks on delete, we set the dialog state appropriately:

```jsx
this.setState({
  dialogState: {
    // ...
  },
});
```

If there is an error, we could do:

```jsx
this.setState({
  ...this.state.dialogState,
  error,
});
```